test/ParseableInterface/synthesized.swift

```diff
@@ -4,20 +4,20 @@
 public enum HasRawValue: Int {
   // CHECK-NEXT: case a, b, c
   case a, b = 5, c
-  // CHECK-NEXT: public typealias RawValue = Swift.Int
-  // CHECK-NEXT: @inlinable public init?(rawValue: Swift.Int)
-  // CHECK-NEXT: public var rawValue: Swift.Int {
-  // CHECK-NEXT:   @inlinable get{{$}}
-  // CHECK-NEXT: }
-} // CHECK-NEXT: {{^}$}}
+  // CHECK-DAG: public typealias RawValue = Swift.Int
+  // CHECK-DAG: @inlinable public init?(rawValue: Swift.Int)
+  // CHECK-DAG: public var rawValue: Swift.Int {
+  // CHECK-DAG:   @inlinable get{{$}}
+  // CHECK-DAG: }
+} // CHECK: {{^}$}}
```

Comment:
This is unfortunate since it doesn't catch that there aren't any *other* members, but I guess it's all right.

Reply:
Could we add CHECK-NOTs after the last CHECK-DAG that would match other declarations, or would that not cover enough possible failures to be useful?

Alternatively, could we make sure synthesized declarations are always listed in the parseable interface in a consistent, non-source-order-related order? That would sidestep the problem entirely.

(Or I could exclude `RawRepresentable` from shuffling, or skip this file during swift_evolve tests. `RawRepresentable` is pretty boring from a resilience standpoint and very interesting in most other respects.)

Reply:
I thought about the CHECK-NOT thing, but it doesn't stop other decls from creeping in *between* the CHECK-DAG lines, or before them.

I think it makes sense for synthesized declarations to be ordered matching the protocol (or, well, type witnesses then non-type witnesses), but I suppose it doesn't so matter so much. I would still want to keep them grouped by protocol, though, so the work to come up with a deterministic order probably isn't worth it.

I'm fine with this solution, or with excluding RawRepresentable from your shuffles, or from skipping this file.

Reply:
cc @harlanhaskins